packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+              disabled={disabled}
+            >
+              {label}
+            </Label>
+          )}
+          <div className={inputWrapperStyles}>
+            <input
+              type={showPassword ? 'text' : 'password'}
+              value={value}
+              id={inputId}
+              autoComplete={autoComplete}
+              aria-describedby={ariaDescribedby}
+              aria-labelledby={ariaLabelledby}
+              aria-disabled={disabled}
+              aria-invalid={state === States.Error || state === States.Warning}
+              className={cx(
+                inputBaseStyles,
+                inputSizeStyles[sizeVariant],
+                inputBaseThemeStyles[theme],
+                {
+                  [inputThemeStyles[theme][state]]: !disabled,
+                  [cx(
+                    inputDisabledBaseStyles,
+                    inputDisabledThemeStyles[theme],
+                  )]: disabled,
+                  [inputIconSizeStyles[sizeVariant]]:
+                    !hasNotifications && state !== States.None,
+                },
+              )}
+              onChange={handleChange}
+              readOnly={disabled ? true : false}
+              {...rest}
+            />
+            {/* Visual icons inside the input will only render if aria-describedby is set and the state is not `none`. `none` does not need a visible icon */}
+            {!hasNotifications && state !== States.None && (
+              <InputIcon state={state} sizeVariant={sizeVariant} />
+            )}
+            <TogglePassword
+              disabled={disabled}
+              showPassword={showPassword}
+              handleTogglePasswordClick={handleTogglePasswordClick}
+              sizeVariant={sizeVariant}
+            />
+          </div>
+          {hasNotifications && (
+            <StateNotifications
+              ariaDescribedby={ariaDescribedby}
+              notifications={stateNotifications as Array<NotificationProps>}
+            />
+          )}
+        </div>
+      </LeafyGreenProvider>
+    );
+  },
+);
+
+PasswordInput.displayName = 'PasswordInput';
+
+const stateNotificationCheck = function (
+  props: { [x: string]: any },
+  propName: string,
+  ...rest: ['componentName', 'location', 'propFullName']
+) {
+  const stateStringProp = PropTypes.oneOf(Object.values(States)).isRequired;
+  const arrayProp = PropTypes.arrayOf(
+    PropTypes.shape({
+      state: PropTypes.oneOf(Object.values(States)).isRequired,
+      notification: PropTypes.string.isRequired,
+    }),
+  );
+  const ariaProp = 'aria-describedby';
+
+  const stateStringType = stateStringProp(props, propName, ...rest);
+  const arrayType = arrayProp(props, propName, ...rest);
+
+  if (typeof props[ariaProp] === 'string') return stateStringType;
+  if (typeof props[ariaProp] === 'undefined') return arrayType;
+
+  return new Error('Error');
+};
+
+PasswordInput.propTypes = {
+  id: PropTypes.string,
+  /// @ts-ignore
```

Comment:
I keep getting these type errors
```
Type 'Requireable<string>' is not assignable to type 'Validator<string>'.
  Types of property '[nominalTypeHack]' are incompatible.
    Type '{ type: string | null | undefined; } | undefined' is not assignable to type '{ type: string; } | undefined'.
      Type '{ type: string | null | undefined; }' is not assignable to type '{ type: string; }'.
        Types of property 'type' are incompatible.
          Type 'string | null | undefined' is not assignable to type 'string'.
            Type 'undefined' is not assignable to type 'string'.

1[8](https://github.com/mongodb/leafygreen-ui/actions/runs/4078324449/jobs/7028400164#step:7:9)1   label: PropTypes.string,
      ~~~~~
```

I have a hunch it's because all the label props are conditional but this error doesn't show up in the `TextInput` component. I added the ignore because its an optional string and consumers will get a warning from the check inside the component if none are used. Have you seen this before?

Reply:
Sean & I found a way around this. I you cast the propTypes object `as unknown` it'll stop interfering with TS types

Reply:
oh! this works but then i get another error, my instinct is to just add a ts-ignore since it shouldn't interfere with TS
![Screen Shot 2023-02-09 at 1 29 38 PM](https://user-images.githubusercontent.com/5740159/217904891-5c35b806-9e6c-4400-bf36-e7a6123105e5.png)